api/ext/tor.py

```diff
@@ -74,26 +85,27 @@ def parse_torrc(torrc):
             )
             services.append(hidden_service)
         elif hidden_service_port and services:
-            services[-1] = services[-1]._replace(port_definition=hidden_service_port)
+            services[-1].port_definition = hidden_service_port
     return services
 
 
+@dataclass
 class TorService:
-    services = []
-    services_dict = {}
-    anonymous_services_dict = {}
-    onion_host = ""
+    services: Optional[list] = None
+    services_dict: Optional[dict] = None
+    anonymous_services_dict: Optional[dict] = None
+    onion_host: Optional[str] = None
 
 
-def refresh():
-    TorService.services = parse_torrc(settings.TORRC_FILE)
-    TorService.services_dict = {service.name: service._asdict() for service in TorService.services}
-    TorService.anonymous_services_dict = {
-        service.name: {"name": service.name, "hostname": service.hostname} for service in TorService.services
+def refresh(torService):
+    torService.services = parse_torrc(settings.TORRC_FILE)
+    torService.services_dict = {service.name: service._asdict() for service in torService.services}
+    torService.anonymous_services_dict = {
+        service.name: {"name": service.name, "hostname": service.hostname} for service in torService.services
     }
-    TorService.onion_host = TorService.services_dict.get("BitcartCC Merchants API", "")
-    if TorService.onion_host:  # pragma: no cover
-        TorService.onion_host = TorService.onion_host["hostname"]
-
+    torService.onion_host = torService.services_dict.get("BitcartCC Merchants API", "")
+    if torService.onion_host:  # pragma: no cover
+        torService.onion_host = torService.onion_host["hostname"]
 
-refresh()
+torService = TorService()
```

Comment:
```suggestion
tor_service = TorService()
```

And in other places. We follow PEP8 naming style:
`some_var` for variables, `SomeVar` for class names, etc.